File: stack,queue,deque/problem_solutions/daily_temperatures.c
#include <stdlib.h>

int* dailyTemperatures(int* temperatures, int temperaturesSize, int* returnSize) {
    int *result = (int*)calloc(temperaturesSize, sizeof(int));
    int *stack = (int*)malloc(temperaturesSize * sizeof(int));
    int length = -1;
    for (int t = 0; t < temperaturesSize; t++)
    {
        while ((length != -1) && (temperatures[t] > temperatures[stack[length]]))
        {
            int previous = stack[length--];
            result[previous] = t - previous; 
        }
        stack[++length] = t;
    }
    free(stack);
    *returnSize = temperaturesSize;
    return (result);
}

File: stack,queue,deque/problem_solutions/daily_temperatures.py
def dailyTemperatures(temperatures):
    output = [0] * len(temperatures)
    stack = []
    for index, temperature in enumerate(temperatures):
        while stack and temperature > stack[-1][0]:
            temp, ind = stack.pop()
            output[ind] = index - ind
        stack.append([temperature, index])
    return (output)